test/Microsoft.AspNetCore.Server.KestrelTests/PipelineExtensionTests.cs

```diff
@@ -0,0 +1,149 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.IO.Pipelines;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Server.Kestrel.Internal.Http;
+using Xunit;
+
+namespace Microsoft.AspNetCore.Server.KestrelTests
+{
+    public class PipelineExtensionTests : IDisposable
```

Comment:
Add a `WriteAsciiNoValidation` test encoding and verifying a single char, with data from 0-255. This guaranteed to character value is special cased in that method.